jax/_src/api_util.py

```diff
@@ -459,24 +459,14 @@ def _flat_out_axes(leaves, treedef, *args, **kwargs):
     raise ValueError(msg) from None
   yield ans, spec_flat
 
-def _isgeneratorfunction(fun):
-  # TODO 3.9+: remove
-  # re-implemented here because of https://bugs.python.org/issue33261
```

Comment:
This bug has been fixes in https://github.com/python/cpython/issues/77442 and `inspect.isgeneratorfunction` uses [this implementation since 3.9](https://github.com/python/cpython/blob/9cf6752276e6fcfd0c23fdb064ad27f448aaaf75/Lib/inspect.py#L172-L188).